### DISCUSSION TITLE
Open dialog widget after page rendering

### DISCUSSION THREAD

**Discussion by adamitj** [primefaces/community#428] [PrimeFaces] [answered]
Kind of silly question, but I'm an outdated Primefaces programmer (used to work with version 8 years ago). I'm trying to open a dialog right after the page is completely rendered, but I'm getting this error:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'show')
    at r2/:6:20
(anonymous) @ r2/:6
```

The page code is this:

```
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://xmlns.jcp.org/jsf/html"
      xmlns:f="http://xmlns.jcp.org/jsf/core"
      xmlns:p="http://primefaces.org/ui">

    <h:head>
    </h:head>
 
    <h:body>
        <p:dialog id="dialog1" header="Header" widgetVar="wdialog1" minHeight="40" width="350" showEffect="fade" modal="true">
            <p>Hello Dialog!</p>
        </p:dialog>
        
        <script>
            var widget = PF('wdialog1');
            widget.show();
        </script>
    </h:body>
</html>
```

What is the best way to open a dialog right after page loads completely? This is only a pretest code, I'll perform other validations to check if the dialog must be opened or not.

**Comment by FlipWarthog** [comment, marked as answer]
I think you just need to wrap the js in jquery's document ready function to defer it:
```js
$( document ).ready(function() {
  var widget = PF('wdialog1');
  widget.show();
});
```

**Reply by adamitj** [reply]
This is exactly what I needed. In fact I was sure enough that it was a _timing_ issue, but wasn't able to find the answer correctly. Thank you very much!

**Reply by FlipWarthog** [reply]
Any time!
